pkg/proxy/ipvs/netlink_linux.go

```diff
@@ -96,3 +104,59 @@ func (h *netlinkHandle) DeleteDummyDevice(devName string) error {
 	}
 	return h.LinkDel(dummy)
 }
+
+// GetLocalAddresses list all LOCAL type IP addresses from host based on filter device.
+// If filter device is not specified, it's equivalent to exec:
+// $ ip route show table local type local proto kernel
+//10.0.0.1 dev kube-ipvs0  scope host  src 10.0.0.1
+//10.0.0.10 dev kube-ipvs0  scope host  src 10.0.0.10
+//10.0.0.252 dev kube-ipvs0  scope host  src 10.0.0.252
+//100.106.89.164 dev eth0  scope host  src 100.106.89.164
+//127.0.0.0/8 dev lo  scope host  src 127.0.0.1
+//127.0.0.1 dev lo  scope host  src 127.0.0.1
+//172.17.0.1 dev docker0  scope host  src 172.17.0.1
+//192.168.122.1 dev virbr0  scope host  src 192.168.122.1
+// Then cut the unique src IP fields,
+// --> result set: [10.0.0.1, 10.0.0.10, 10.0.0.252, 100.106.89.164, 127.0.0.1, 192.168.122.1]
+
+// If filter device is specified, it's equivalent to exec:
+// $ ip route show table local type local proto kernel dev kube-ipvs0
+// 10.0.0.1  scope host  src 10.0.0.1
+// 10.0.0.10  scope host  src 10.0.0.10
+// Then cut the unique src IP fields,
+// --> result set: [10.0.0.1, 10.0.0.10]
+func (h *netlinkHandle) GetLocalAddresses(filterDev string) (sets.String, error) {
```

Comment:
Where are the tests?

Reply:
Believe me, I am 100% want to add tests.

The reason why I didn't do that because I am unable to do that.  `GetLocalAddresses()` will revoke `netlink.LinkByName()` to get a REAL network interface and `netlink.RouteListFiltered()` to list REAL route table.  In order to testing them, I have to create a REAL network interface and give the test code root privilege.  Unfortunately, the UT codes run in upstream CI can NOT acquire such privileges. Am I missing something?

Or, Should we check the user id before running tests? e.g.

```
if user.Current().Uid == 0 {
  do test
}
```

??

It seems odd.

Reply:
In my view, we can add tests for the caller side: `NodeIPs()`?